### PR TITLE
PRO-2466 Documents the http module

### PR DESCRIPTION
**Pull request by abea**
## Summary

PRO-2466

## What are the specific steps to test this change?

Review for accuracy and clarity

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [x] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [x] Related documentation has been updated
- [x] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: docs/reference/modules/http.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+---
+extends: '@apostrophecms/module'
+---
+
+# `@apostrophecms/http`
+
+**Alias:** `apos.http`
+
+This module adds methods for easy HTTP requests in Apostrophe project and module server-side code. It also defines error codes for use with the main `apos.error()` method.
```
Comment (boutell) [inline docs/reference/modules/http.md:9]: It should be mentioned here that these methods are deliberately compatible with those available on the browser side for developer convenience.

File: docs/reference/modules/http.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+## Featured methods
+
+The following methods belong to this module and may be useful in project-level code. See the [source code](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/http/index.js) for all methods that belong to this module.
+<!-- Some are used within the module and would just create noise here. -->
+
+Because this module has an alias, you can call these from another module from the alias path. For example, `self.apos.http.addError()`.
+
+### `addError(name, code)`
+
+An alternate way to register error names to HTTP error codes. Pass the semantic name as the first argument, followed by the `code` integer.
```
Comment (boutell) [inline docs/reference/modules/http.md:65]: ... which becomes the HTTP status code if that error is thrown in an Apostrophe route via `throw self.apos.error('errorname')`.

File: docs/reference/modules/http.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+### `async get(url, options)`
+
+Fetch the given URL (`url`) via a `GET` request and return the response body. If the URL is relative (starts with `/`) it will be requested from the Apostrophe site itself. This is helpful to avoid needing to construct the base URL across environments.
+
+Accepts the following properties on the optional `options` object:
+
+| Option | Type |Description |
+| -------- | ------ | ----------- |
+| `qs` | Object | An object of query parameters and values. |
```
Comment (boutell) [inline docs/reference/modules/http.md:75]: May contain nested objects, arrays, and properties with `null` values.

File: docs/reference/modules/http.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+Send a `POST` request to the given URL and return the response body. If the URL is relative (starts with `/`) it will be requested from the Apostrophe site itself. This is helpful to avoid needing to construct the base URL across environments.
+
+Accepts the following properties on the optional `options` object:
+
+| Option | Type |Description |
+| -------- | ------ | ----------- |
+| `qs` | Object | An object of query parameters and values. |
+| `jar` | Object | Pass in a cookie jar obtained from `apos.http.jar()`. |
+| `send` | String | Can be 'json' to always send `options.body` JSON-encoded, or `form` to send it URL-encoded. |
+| `body` | Variable | The request body. If an object or array, it is sent as JSON. Otherwise it is sent as-is, unless the `send` option is set. |
```
Comment (boutell) [inline docs/reference/modules/http.md:111]: If it is a `FormData` object constructed via the `form-data` npm module, it is sent as a `multipart/form-data` upload, in order to match the browser-side support for file uploads.

**Review by boutell — CHANGES_REQUESTED** [review]
This is great. A few things.

File: docs/reference/modules/http.md [hunk 5]
```diff
@@ -105,10 +106,10 @@ Accepts the following properties on the optional `options` object:
 
 | Option | Type |Description |
 | -------- | ------ | ----------- |
-| `qs` | Object | An object of query parameters and values. |
+| `qs` | Object | An object of query parameters and values. It may contain nested objects, arrays, and properties with null values. |
 | `jar` | Object | Pass in a cookie jar obtained from `apos.http.jar()`. |
 | `send` | String | Can be 'json' to always send `options.body` JSON-encoded, or `form` to send it URL-encoded. |
-| `body` | Variable | The request body. If an object or array, it is sent as JSON. Otherwise it is sent as-is, unless the `send` option is set. |
+| `body` | Variable | The request body. If an object or array, it is sent as JSON. If a `FormData` object, it is sent as a `multipart/form-data` upload. Otherwise it is sent as-is, unless the `send` option is set. |
```
Comment (boutell) [inline docs/reference/modules/http.md:112]: `FormData` doesn't just exist in server land, so this should link specifically to:

https://www.npmjs.com/package/form-data

Which is known to be compatible.

**Review by boutell — CHANGES_REQUESTED** [review]
One tiiiiny thing

File: docs/reference/api/media.md [hunk 6]
```diff
@@ -19,12 +19,12 @@ For attachments meant to be included in the media or file libraries, metadata an
 
 **Authentication required.**
 
-The `body` of the request should use the `multipart/form-data` encoding, with the file itself uploaded under the name `file`. In a client-side request, the `body` of the request should be a `FormData` object.
+The `body` of the request should use the `multipart/form-data` encoding, with the file itself uploaded under the name `file`. In a client-side request, the `body` of the request should be a [`FormData` object](https://www.npmjs.com/package/form-data).
```
Comment (boutell) [inline docs/reference/api/media.md:22]: Unfortunately this is reversed. `FormData` is always a thing in the browser. The link is for the server side module's use of it.

File: docs/reference/api/media.md [hunk 7]
```diff
@@ -19,12 +19,12 @@ For attachments meant to be included in the media or file libraries, metadata an
...
 
 ### Request example
 
 ```javascript
-// Create a `FormData` object
+// Create a [`FormData` object](https://www.npmjs.com/package/form-data)
```
Comment (boutell) [inline docs/reference/api/media.md:27]: This link does not belong in browser side code.

File: docs/reference/modules/http.md [hunk 8]
```diff
@@ -149,7 +149,7 @@ Accepts the following properties on the optional `options` object:
 | `qs` | Object | An object of query parameters and values. It may contain nested objects, arrays, and properties with null values. |
 | `jar` | Object | Pass in a cookie jar obtained from `apos.http.jar()`. |
 | `send` | String | Can be 'json' to always send `options.body` JSON-encoded, or `form` to send it URL-encoded. |
-| `body` | Variable | The request body. If an object or array, it is sent as JSON. If a `FormData` object, it is sent as a `multipart/form-data` upload. Otherwise it is sent as-is, unless the `send` option is set. |
+| `body` | Variable | The request body. If an object or array, it is sent as JSON. If a [`FormData` object](https://www.npmjs.com/package/form-data), it is sent as a `multipart/form-data` upload. Otherwise it is sent as-is, unless the `send` option is set. |
```
Comment (boutell) [inline docs/reference/modules/http.md:152]: The link does belong here.

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/reference/api/media.md (hunk 7 above)
Comment (abea) [reply docs/reference/api/media.md:27]: bad replace-all-jitsu.

**Review by boutell — APPROVED** [review]

**Pull request merged**